lib/agenda.js

```diff
@@ -328,7 +329,7 @@ Agenda.prototype.saveJob = function(job, cb) {
       update = { $set: props };
 
   if (id) {
-    this._collection.findAndModify({_id: id}, {}, update, {new: true}, processDbResult );
+    this._collection.findOneAndUpdate({_id: id}, update, {returnOriginal: false}, processDbResult );
```

Comment:
I can't find any reference to a `returnOriginal` option for `findOneAndUpdate`. Did you mean to use `returnNewDocument`?